File: net/JetBrains.Serialization/src/Dataholders/RsaEncapContentInfo.cs
using Newtonsoft.Json;
using Org.BouncyCastle.Asn1;
using Org.BouncyCastle.Asn1.Cms;

namespace JetBrains.Serialization;

[JsonObject(MemberSerialization.OptIn)]
public class RsaEncapContentInfo : EncapContentInfo
{
  [JsonProperty("ContentType")] protected override TextualInfo ContentType { get; }
  [JsonProperty("Content")] private TextualInfo? _content;

  [JsonConstructor]
  public RsaEncapContentInfo(TextualInfo contentType, TextualInfo? content)
  {
    ContentType = contentType;
    _content = content;
  }

  public RsaEncapContentInfo(ContentInfo contentInfo)
    : this(TextualInfo.GetInstance(contentInfo.ContentType),
      contentInfo.Content != null ? TextualInfo.GetInstance(contentInfo.Content) : null)
  {
  }

  protected override Asn1Encodable? GetContentPrimitive() => _content?.ToPrimitive();
}

File: jvm/src/main/kotlin/com/jetbrains/signatureverifier/serialization/dataholders/RsaEncapContentInfo.kt
package com.jetbrains.signatureverifier.serialization.dataholders

import kotlinx.serialization.Serializable
import org.bouncycastle.asn1.cms.ContentInfo

@Serializable
data class RsaEncapContentInfo(
  override val contentType: TextualInfo,
  val content: TextualInfo?
) : EncapContentInfo() {

  constructor(contentInfo: ContentInfo) : this(
    TextualInfo.getInstance(contentInfo.contentType),
    contentInfo.content?.let { TextualInfo.getInstance(it) }
  )

  override fun getContentPrimitive() =
    content?.toPrimitive()
}